lib/updater.ex

```diff
@@ -0,0 +1,116 @@
+defmodule Elixometer.Updater do
+  @moduledoc """
+  A capped worker that updates metrics.
+  """
+
+  import Elixometer, only: [ensure_registered: 2, add_counter: 2]
+  import Elixometer.Utils
+  use GenServer
+
+  def init([]) do
+    {:ok, pobox} = :pobox.start_link(self, 1000, :queue)
+    Process.register(pobox, :elixometer_pobox)
+    activate_pobox
+    {:ok, nil}
+  end
+
+  def start_link do
+    GenServer.start_link(__MODULE__, [], name: __MODULE__)
+  end
+
+  def timer(name, units, elapsed) when is_bitstring(name) do
+    elixometer_name = name_to_exometer(:timers, name)
+    timer(elixometer_name, units, elapsed)
+  end
+
+  def timer(name, units, elapsed) when is_list(name) do
+  :pobox.post(:elixometer_pobox, {:timer, name, units, elapsed})
+  end
+
+  def gauge(name, value) do
+    :pobox.post(:elixometer_pobox, {:gauge, name, value})
+  end
+
+  def counter(name, delta, reset_seconds) do
+    :pobox.post(:elixometer_pobox, {:counter, name, delta, reset_seconds})
+  end
+
+  def spiral(name, delta, opts) do
+    :pobox.post(:elixometer_pobox, {:spiral, name, delta, opts})
+  end
+
+  def histogram(name, delta, reset_seconds) do
+    :pobox.post(:elixometer_pobox, {:histogram, name, delta, reset_seconds})
+  end
+
+  def handle_info({:mail, _pid, messages, _, _}, state) do
+    messages
+    |> Enum.each(&do_update/1)
```

Comment:
I like `Enum.each(messages, &do_update/1)` for these simple calls, but it's totally fine either way.